### DISCUSSION TITLE
Focus follows the mouse but ONLY in the visible windows (ones in current column)!

### DISCUSSION THREAD

**Discussion by rempas** [YaLTeR/niri#408] [Ideas]
The current focus behavior when following the mouse is, to be able to change to non-visible windows by moving the mouse to the top left or top right corners of the screen.

This can mess some windows and make them more hard to use when you want to have a maximized column. One example been the current one that made my post this issue, LMMS.

**Comment by YaLTeR** [comment]
This is not a hardcoded behavior, just a result of how it currently works. Focus follows mouse will change the focus when a different window gets the pointer focus, and windows generally have a small area extending around them where they can get pointer focus (for example, for CSD resizing).

**Reply by rempas** [reply]
Thanks for the information! Does it make sense however if they didn't, and they were organized based on what you see?

**Reply by YaLTeR** [reply]
Not sure. Maybe it would make more sense to keep the input region check for the currently focused window (so that you can move the mouse out and resize from CSD) but make the switch-focus check use the window geometry? There may be some edge cases that I can't think of right now. For one, hovering over pop-up menus extending outside the window won't focus it this way.

**Reply by rempas** [reply]
Exactly, good call! Well, I think my "job" here is done. It's up to you to decide what you want to do ;)

As always, thank you for your work!

**Comment by YaLTeR** [comment]
Let's discuss in https://github.com/YaLTeR/niri/discussions/491
